src/CoreShop/Bundle/OrderBundle/Controller/OrderCommentController.php

```diff
@@ -0,0 +1,130 @@
+    {
+        $orderId = $request->get('id');
+        $order = $this->getOrderRepository()->find($orderId);
+
+        $list = [];
+        $objectNoteService = $this->get('coreshop.object_note_service');
+        $notes = $objectNoteService->getObjectNotes($order, Notes::NOTE_ORDER_COMMENT);
+
+        $parsedData = [];
+        /** @var Note $note */
+        foreach ($notes as $note) {
+            $user = User::getById($note->getUser());
+            $noteData = $note->getData();
+            $parsedData[] = [
+                'id'            => $note->getId(),
+                'text'          => $note->getDescription(),
+                'date'          => $note->getDate(),
+                'userName'      => $user ? $user->getName() : 'anonymous',
+                'submitAsEmail' => isset($noteData['submitAsEmail']) && $noteData['submitAsEmail']['data'] === true
+            ];
+        }
+
+        return $this->viewHandler->handle(['success' => true, 'comments' => $parsedData]);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return \Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse
+     */
+    public function addAction(Request $request)
+    {
+        $comment = $request->get('comment');
+        $submitAsEmail = $request->get('submitAsEmail') === 'true';
+        $orderId = $request->get('id');
+
+        $order = $this->getOrderRepository()->find($orderId);
+
+        if (!$order instanceof OrderInterface) {
+            return $this->viewHandler->handle(['success' => false, 'message' => "Order with ID '$orderId' not found"]);
+        }
+
+        try {
+
+            $objectNoteService = $this->get('coreshop.object_note_service');
+            $commentEntity = $objectNoteService->createPimcoreNoteInstance($order, Notes::NOTE_ORDER_COMMENT);
+            $commentEntity->setTitle('Order Comment');
+            $commentEntity->setDescription(nl2br($comment));
+            $commentEntity->addData('submitAsEmail', 'bool', $submitAsEmail);
+            $comment = $objectNoteService->storeNote($commentEntity, $emailDocument);
+
+            $this->get('event_dispatcher')->dispatch(
+                'coreshop.comment.order_comment_post_add',
+                new GenericEvent($comment, ['order' => $order, 'submitAsEmail' => $submitAsEmail])
+            );
+
+            return $this->viewHandler->handle(['success' => true, 'commentId' => $comment->getId()]);
+        } catch (\Exception $ex) {
+            return $this->viewHandler->handle(['success' => false, 'message' => $ex->getMessage()]);
+        }
+    }
+
+    /**
+     * @param Request $request
+     * @return mixed
+     */
+    public function deleteAction(Request $request)
+    {
+        $commentId = $request->get('id');
+        $objectNoteService = $this->get('coreshop.object_note_service');
+        $commentEntity = $objectNoteService->getNoteById($commentId);
+        $commentEntity->delete();
+
+        $this->get('event_dispatcher')->dispatch(
```

Comment:
As discussed, please do this inside the coreshop.object_note_service. Therefore, also add a delete function there.